### ISSUE TITLE
Video Playback Speed linked to Size of Video (Resolution)

### ISSUE THREAD

**Issue by ghost** [huskeee/tkvideo#10]
So I've built tkvideo into my project, got my video playing.
I didn't want my video to be 1280x720, so I changed the resolution to 600x400.

**What happened:** Video Playback got really fast. When I altered the reolution to 1920x1080, i noticed that the Video would play slower.

Heres my code, I basically just altered the resolution, changed the label to my label's name ands changed the videopath & name:

`player = tkvideo("./video.mp4", label, loop = 1, size = (600,400))`

I used a Music Video in 480p. tried various Resolutions and they either started in sync and got faster, started slower and got faster, or were just completely off in terms of playback speed

**Comment by Gogotron** [comment]
Is this not the same as issue #1 which was closed by pull request #8?

**Comment by Z1YAD-M** [comment]
From what I found, the module in my libraries is not updated to pull request #8. Even after updating it myself, I did not find the "hz" parameter to have any effect on increasing FPS past a certain point. Whether I gave the video its original resolution or down scaled it, FPS seems to auto cap at around 15-20. I set the hz parameter to 999999999 and there was no increase. I set it to 1, and it got infinitely slower, as if it was only displaying an image. It seems like the issue is the values have incorrectly assigned meanings. The resolution in my case makes no difference

**Comment by Z1YAD-M** [comment]
@tobiglaser

**Comment by tobiglaser** [comment]
PR #8 was merely a fix for the video playing too fast.
It's not possible to speed it up with the parameter.
The maximum speed is dictated by the system reading and parsing each frame as a single image.
This, like missing sound, is sadly inherent to the way this project works.
Maybe the parameter should have been `fps_limit` rather then `hz`.
The project not being updated on [pypi](https://pypi.org/project/tkVideo/#history) is a different story though.

**Comment by Z1YAD-M** [comment]
@tobiglaser  BTW I am trying to make it possible to delete the selected playing video from remaining on playing on the tkinter window. In an if,elif,else statement setup, I want the video to be deleted and replaced with the output of another statement when necessary (i am trying to achieve this). But I found out the module itself doesnt have the attribute destroy. Do you think you can help add a destroy function? Or give hints as to how to add it? (beginner here)

**Comment by Z1YAD-M** [comment]
You and I are the only guys that seem to still be alive on this module page haha

**Comment by tobiglaser** [comment]
Seems so. Some thoughts:
- This project is unsuitable for any intent bigger then displaying a gif.
- The approach to displaying moving images has been flawed from the start.
- The naming of this package has been misleading python users since it's release.
- The prefix tk implies the ease of use and usability Tk is known for.
- This is the reason so many folks come here asking for audio and other features.

I ended up here like this as well.
For suffisticated video displaying a proper video player can and should be used.
For the project I was working on at the time this is [`python-vlc`](https://pypi.org/project/python-vlc/).
There is an example hidden [here](https://git.videolan.org/?p=vlc/bindings/python.git;a=blob;f=examples/tkvlc.py;h=9984138afa37132ad1279e55d66eb7b705e21b98;hb=HEAD). Not as user friendly, but ey, you want to be a real programmer with nice video in your juicy GUI, you gotta level up your programming.

Ok, after getting that of my chest:
Since I commited #8 I found an even better and much simpler way of getting the same functionality even _without threading_!
It uses the timed callbacks built into the Tk framework. Something like `root.after(frametime, root.nextFrame)`.
I've seen it implemented for clocks and even for video.
As you might have guessed I'm not looking to actively work on tkVideo at this time.

I'm not sure about what you mean by "destroying". Have a look into PR #5. After stopping I would try replacing the whole label with a different one (or something like that).
There should be some way to hide a label, place yours next to it and hide one or the other.
